### ISSUE TITLE
[Bug] The performance of h3dnet on SUNRGBD is far low than their paper reported

### ISSUE THREAD

**Issue by EmiyaNing** [open-mmlab/mmdetection3d#2772]
### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmdetection3d/issues) and [Discussions](https://github.com/open-mmlab/mmdetection3d/discussions) but cannot get the expected help.
- [X] I have read the [FAQ documentation](https://mmdetection3d.readthedocs.io/en/latest/notes/faq.html) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version (dev-1.x)](https://github.com/open-mmlab/mmdetection3d/tree/dev-1.x) or [latest version (dev-1.0)](https://github.com/open-mmlab/mmdetection3d/tree/dev-1.0).

### Task

I have modified the scripts/configs, or I'm working on my own tasks/models/datasets.

### Branch

main branch https://github.com/open-mmlab/mmdetection3d

### Environment

sys.platform: linux
Python: 3.8.10 (default, Jun  4 2021, 15:09:15) [GCC 7.5.0]
CUDA available: True
numpy_random_seed: 2147483648
GPU 0,1: NVIDIA GeForce RTX 3090
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 11.1, V11.1.105
GCC: gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
PyTorch: 1.9.0+cu111
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.1.2 (Git Hash 98be7e8afa711dc9b66c8ff3504129cb82013cdb)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86
  - CuDNN 8.0.5
  - Magma 2.5.2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.1, CUDNN_VERSION=8.0.5, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.9.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON,

TorchVision: 0.10.0+cu111
OpenCV: 4.8.0
MMEngine: 0.8.4
MMDetection: 3.1.0
MMDetection3D: 1.2.0+36758c0
spconv2.0: True

### Reproduces the problem - code sample

_base_ = [
    '../_base_/datasets/sunrgbd-3d.py', '../_base_/schedules/schedule-3x.py', 
    '../_base_/default_runtime.py', '../_base_/models/h3dnet.py'
]

primitive_z_cfg = dict(
    type='PrimitiveHead',
    num_dims=2,
    num_classes=10,
    primitive_mode='z',
    upper_thresh=100.0,
    surface_thresh=0.5,
    vote_module_cfg=dict(
        in_channels=256,
        vote_per_seed=1,
        gt_per_seed=1,
        conv_channels=(256, 256),
        conv_cfg=dict(type='Conv1d'),
        norm_cfg=dict(type='BN1d'),
        norm_feats=True,
        vote_loss=dict(
            type='ChamferDistance',
            mode='l1',
            reduction='none',
            loss_dst_weight=10.0)),
    vote_aggregation_cfg=dict(
        type='PointSAModule',
        num_point=1024,
        radius=0.3,
        num_sample=16,
        mlp_channels=[256, 128, 128, 128],
        use_xyz=True,
        normalize_xyz=True),
    feat_channels=(128, 128),
    conv_cfg=dict(type='Conv1d'),
    norm_cfg=dict(type='BN1d'),
    objectness_loss=dict(
        type='mmdet.CrossEntropyLoss',
        class_weight=[0.4, 0.6],
        reduction='mean',
        loss_weight=30.0),
    center_loss=dict(
        type='ChamferDistance',
        mode='l1',
        reduction='sum',
        loss_src_weight=0.5,
        loss_dst_weight=0.5),
    semantic_reg_loss=dict(
        type='ChamferDistance',
        mode='l1',
        reduction='sum',
        loss_src_weight=0.5,
        loss_dst_weight=0.5),
    semantic_cls_loss=dict(
        type='mmdet.CrossEntropyLoss', reduction='sum', loss_weight=1.0),
    train_cfg=dict(
        sample_mode='vote',
        dist_thresh=0.2,
        var_thresh=1e-2,
        lower_thresh=1e-6,
        num_point=100,
        num_point_line=10,
        line_thresh=0.2),
    test_cfg=dict(sample_mode='seed'))

primitive_xy_cfg = dict(
    type='PrimitiveHead',
    num_dims=2,
    num_classes=10,
    primitive_mode='xy',
    upper_thresh=100.0,
    surface_thresh=0.5,
    vote_module_cfg=dict(
        in_channels=256,
        vote_per_seed=1,
        gt_per_seed=1,
        conv_channels=(256, 256),
        conv_cfg=dict(type='Conv1d'),
        norm_cfg=dict(type='BN1d'),
        norm_feats=True,
        vote_loss=dict(
            type='ChamferDistance',
            mode='l1',
            reduction='none',
            loss_dst_weight=10.0)),
    vote_aggregation_cfg=dict(
        type='PointSAModule',
        num_point=1024,
        radius=0.3,
        num_sample=16,
        mlp_channels=[256, 128, 128, 128],
        use_xyz=True,
        normalize_xyz=True),
    feat_channels=(128, 128),
    conv_cfg=dict(type='Conv1d'),
    norm_cfg=dict(type='BN1d'),
    objectness_loss=dict(
        type='mmdet.CrossEntropyLoss',
        class_weight=[0.4, 0.6],
        reduction='mean',
        loss_weight=30.0),
    center_loss=dict(
        type='ChamferDistance',
        mode='l1',
        reduction='sum',
        loss_src_weight=0.5,
        loss_dst_weight=0.5),
    semantic_reg_loss=dict(
        type='ChamferDistance',
        mode='l1',
        reduction='sum',
        loss_src_weight=0.5,
        loss_dst_weight=0.5),
    semantic_cls_loss=dict(
        type='mmdet.CrossEntropyLoss', reduction='sum', loss_weight=1.0),
    train_cfg=dict(
        sample_mode='vote',
        dist_thresh=0.2,
        var_thresh=1e-2,
        lower_thresh=1e-6,
        num_point=100,
        num_point_line=10,
        line_thresh=0.2),
    test_cfg=dict(sample_mode='seed'))

primitive_line_cfg = dict(
    type='PrimitiveHead',
    num_dims=2,
    num_classes=10,
    primitive_mode='line',
    upper_thresh=100.0,
    surface_thresh=0.5,
    vote_module_cfg=dict(
        in_channels=256,
        vote_per_seed=1,
        gt_per_seed=1,
        conv_channels=(256, 256),
        conv_cfg=dict(type='Conv1d'),
        norm_cfg=dict(type='BN1d'),
        norm_feats=True,
        vote_loss=dict(
            type='ChamferDistance',
            mode='l1',
            reduction='none',
            loss_dst_weight=10.0)),
    vote_aggregation_cfg=dict(
        type='PointSAModule',
        num_point=1024,
        radius=0.3,
        num_sample=16,
        mlp_channels=[256, 128, 128, 128],
        use_xyz=True,
        normalize_xyz=True),
    feat_channels=(128, 128),
    conv_cfg=dict(type='Conv1d'),
    norm_cfg=dict(type='BN1d'),
    objectness_loss=dict(
        type='mmdet.CrossEntropyLoss',
        class_weight=[0.4, 0.6],
        reduction='mean',
        loss_weight=30.0),
    center_loss=dict(
        type='ChamferDistance',
        mode='l1',
        reduction='sum',
        loss_src_weight=1.0,
        loss_dst_weight=1.0),
    semantic_reg_loss=dict(
        type='ChamferDistance',
        mode='l1',
        reduction='sum',
        loss_src_weight=1.0,
        loss_dst_weight=1.0),
    semantic_cls_loss=dict(
        type='mmdet.CrossEntropyLoss', reduction='sum', loss_weight=2.0),
    train_cfg=dict(
        sample_mode='vote',
        dist_thresh=0.2,
        var_thresh=1e-2,
        lower_thresh=1e-6,
        num_point=100,
        num_point_line=10,
        line_thresh=0.2),
    test_cfg=dict(sample_mode='seed'))

# model settings
model = dict(
    rpn_head=dict(
        num_classes=10,
        bbox_coder=dict(
            type='PartialBinBasedBBoxCoder',
            num_sizes=10,
            num_dir_bins=12,
            with_rot=True,
            mean_sizes=[
                [2.114256, 1.620300, 0.927272], [0.791118, 1.279516, 0.718182],
                [0.923508, 1.867419, 0.845495], [0.591958, 0.552978, 0.827272],
                [0.699104, 0.454178, 0.75625], [0.69519, 1.346299, 0.736364],
                [0.528526, 1.002642, 1.172878], [0.500618, 0.632163, 0.683424],
                [0.404671, 1.071108, 1.688889], [0.76584, 1.398258, 0.472728]])),
    roi_head=dict(
        bbox_head=dict(
            num_classes=10,
            bbox_coder=dict(
                type='PartialBinBasedBBoxCoder',
                num_sizes=10,
                num_dir_bins=12,
                with_rot=True,
                mean_sizes=[
                [2.114256, 1.620300, 0.927272], [0.791118, 1.279516, 0.718182],
                [0.923508, 1.867419, 0.845495], [0.591958, 0.552978, 0.827272],
                [0.699104, 0.454178, 0.75625], [0.69519, 1.346299, 0.736364],
                [0.528526, 1.002642, 1.172878], [0.500618, 0.632163, 0.683424],
                [0.404671, 1.071108, 1.688889], [0.76584, 1.398258, 0.472728]]))))

train_dataloader = dict(
    batch_size=4,
    num_workers=2,
)

# yapf:disable
# yapf:enable
# Default setting for scaling LR automatically
#   - `enable` means enable scaling LR automatically
#       or not by default.
#   - `base_batch_size` = (8 GPUs) x (3 samples per GPU).
auto_scale_lr = dict(enable=False, base_batch_size=24)

### Reproduces the problem - command or script

./tools/dist_train.sh configs/h3dnet/h3dnet_sunrgbd_3d.py

### Reproduces the problem - error message

+-------------+---------+---------+---------+---------+
| classes     | AP_0.25 | AR_0.25 | AP_0.50 | AR_0.50 |
+-------------+---------+---------+---------+---------+
| bed         | 0.8229  | 0.9282  | 0.2197  | 0.4175  |
| table       | 0.4265  | 0.8169  | 0.1303  | 0.3292  |
| sofa        | 0.5789  | 0.8804  | 0.3018  | 0.5152  |
| chair       | 0.7824  | 0.8978  | 0.5667  | 0.6779  |
| toilet      | 0.8894  | 0.9724  | 0.5962  | 0.7379  |
| desk        | 0.1897  | 0.7572  | 0.0195  | 0.2046  |
| dresser     | 0.2849  | 0.7339  | 0.1236  | 0.3440  |
| night_stand | 0.5700  | 0.9137  | 0.2852  | 0.5765  |
| bookshelf   | 0.1358  | 0.4894  | 0.0083  | 0.0993  |
| bathtub     | 0.7103  | 0.8980  | 0.1619  | 0.3878  |
+-------------+---------+---------+---------+---------+
| Overall     | 0.5391  | 0.8288  | 0.2413  | 0.4290  |
+-------------+---------+---------+---------+---------+
2023/10/14 23:59:48 - mmengine - INFO - Epoch(val) [36][2525/2525]    bed_AP_0.25: 0.8229  table_AP_0.25: 0.4265  sofa_AP_0.25: 0.5789  chair_AP_0.25: 0.7824  toilet_AP_0.25: 0.8894  desk_AP_0.25: 0.1897  dresser_AP_0.25: 0.2849  night_stand_AP_0.25: 0.5700  bookshelf_AP_0.25: 0.1358  bathtub_AP_0.25: 0.7103  mAP_0.25: 0.5391  bed_rec_0.25: 0.9282  table_rec_0.25: 0.8169  sofa_rec_0.25: 0.8804  chair_rec_0.25: 0.8978  toilet_rec_0.25: 0.9724  desk_rec_0.25: 0.7572  dresser_rec_0.25: 0.7339  night_stand_rec_0.25: 0.9137  bookshelf_rec_0.25: 0.4894  bathtub_rec_0.25: 0.8980  mAR_0.25: 0.8288  bed_AP_0.50: 0.2197  table_AP_0.50: 0.1303  sofa_AP_0.50: 0.3018  chair_AP_0.50: 0.5667  toilet_AP_0.50: 0.5962  desk_AP_0.50: 0.0195  dresser_AP_0.50: 0.1236  night_stand_AP_0.50: 0.2852  bookshelf_AP_0.50: 0.0083  bathtub_AP_0.50: 0.1619  mAP_0.50: 0.2413  bed_rec_0.50: 0.4175  table_rec_0.50: 0.3292  sofa_rec_0.50: 0.5152  chair_rec_0.50: 0.6779  toilet_rec_0.50: 0.7379  desk_rec_0.50: 0.2046  dresser_rec_0.50: 0.3440  night_stand_rec_0.50: 0.5765  bookshelf_rec_0.50: 0.0993  bathtub_rec_0.50: 0.3878  mAR_0.50: 0.4290  data_time: 0.0019  time: 0.1363

### Additional information

1. We modify the h3dnet_sunrgbd_3d.py from h3dnet_8xb3_scannet_seg.py, and train it with two RTX 3090 GPU. The performance of this model is far lower than their author reproted in their paper.
